### PR TITLE
[#2608] add name to specific links

### LINKED ISSUE #2608: Add Multimedia Name and Use Cases Name to Resource and Provider (Resource, Provider Profile v4.0)
According to the Profile 4.0 add two new fields to the Marketing section for Resource. 
https://confluence.egi.eu/display/EOSCEN/EOSC+Resource+Profile+v4.00
image

According to the Profile 4.0 add one new field to the Marketing section for Provider.
https://confluence.egi.eu/display/EOSCEN/EOSC+Provider+Profile+v4.00
image

Check the correct integration with the PC provider. 
PC add this field to https://beta.providers.eosc-portal.eu.
Actual situation
PC beta Provider EGI Fundation:
image
Master MP
image
image

PC beta Resource EGI Check-In
image
Master MP
image

For Providers, the validation error does not cause the lack of updates???

### PR DESCRIPTION
**Pull request by goreck888**
Prepare for the Resource and Provider Profile 4.0

Closes #2608

### CODE REVIEW

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Steps to reproduce
1. Go to /backoffice/services/new
2. Fill form, add multimedia url, use case url
image
3. Click button "Create resource"
4. Go to Details - see error 
image

**Comment by Marcelinna** [comment]
image
Change to "Url can't be blank"

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Steps to reproduce
1. Go to /backoffice/services/new
2. Fill form, add multimedia url, use case url
image
3. Click button "Preview"
4. Go to Details - see error 
image

**Comment by Marcelinna** [comment]
Go to /backoffice/providers/new, make element Multimedia urls expanded
image

**Comment by Marcelinna** [comment]
Step to reproduce 
1. Add 3 Multimedia urls and 3 use case url (don't fill all required fields)
image
2. Click button "Create resource" - see error - second urls disappear
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Please add information 
Url should start with http:// or https:// prefix [e.g. https://example.com] 
image
just like here
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Step to reproduce
1. Go to /backoffice/services/new
2. Fill required field, add Multimedia urls and Use cases urls 
3. image
4. Click the button "Create resource"
image
5. Click the button "Edit resource"
6. Delete a value from Multimedia urls and Use cases urls 
image
7. Click the button "Update resource"
8. See error - Multimedia nad Use cases are visible in Details, despite fields are empty
image
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Steps to reproduce: 

1. /backoffice/providers/new
2. Fill required fields and Multimedia urls 
image
4. Click button "Create Provider"
image
6. Click the button "Edit"
7. Delete value from Multimedia urls 
image
8. Click the button "Update provider" - see error - no possible update provider 
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Step to reproduce
1. Create provider in beta.provider and add Multimedia URL 
2. Go to provider in Marketplace
3. See error - Multimedia Url doesn't visible
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Steps to reproduce
1. Go to Provider Portal
2. Create Providers and add Multimedia URL, Multimedia Name
3. Go to MP - multimedia URL is visible properly
image
4. Go to Provider Portal and delete Multimedia URL, Multimedia Name
5. See error - link is not removed from MP
image

**Review by Marcelinna — APPROVED** [review]

File: spec/factories/links/link_multimedia_urls.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :link_multimedia_url, class: Link::MultimediaUrl do
+    sequence(:name) { |n| "Multimedia #{n}" }
+    sequence(:url) { |n| "http://example#{n}.org" }
```
Comment (kmarszalek) [inline spec/factories/links/link_multimedia_urls.rb:6]: exampleN.org will be unavailable for N > 3

File: spec/factories/links/link_use_cases_urls.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :link_use_cases_url, class: Link::UseCasesUrl do
+    sequence(:name) { |n| "Use case #{n}" }
+    sequence(:url) { |n| "http://example#{n}.org" }
```
Comment (kmarszalek) [inline spec/factories/links/link_use_cases_urls.rb:6]: exampleN.org will be unavailable for N > 3

File: db/migrate/20220426103925_move_links_to_links_table.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class MoveLinksToLinksTable < ActiveRecord::Migration[6.1]
+  def change
```
Comment (kmarszalek) [inline db/migrate/20220426103925_move_links_to_links_table.rb:4]: is it easily reversibly or do you need up/down ?

File: app/views/components/presentable/details_component/_links.html.haml [hunk 4]
```diff
@@ -5,8 +5,7 @@
       - object.send(field).each_with_index do |link, idx|
         %li.links
           %i.fas.fa-arrow-right
-          -# TODO: refactor dynamic translation
-          = link_to(t(".#{field}", idx: idx + 1), link, "data-probe": "",
+          = link_to(link.name? ? link.name : t(".#{field}", idx: idx + 1), link.url, "data-probe": "",
           "data-target": local_assigns[:preview] ? "preview.link" : "")
```
Comment (kmarszalek) [inline app/views/components/presentable/details_component/_links.html.haml:9]: shouldn't there also be a comment like in sidebar_component _links ? about refactor?

File: app/views/components/presentable/details_component/_links.html.haml [hunk 5]
```diff
@@ -5,8 +5,7 @@
       - object.send(field).each_with_index do |link, idx|
         %li.links
           %i.fas.fa-arrow-right
-          -# TODO: refactor dynamic translation
-          = link_to(t(".#{field}", idx: idx + 1), link, "data-probe": "",
+          = link_to(link.name? ? link.name : t(".#{field}", idx: idx + 1), link.url, "data-probe": "",
```
Comment (kmarszalek) [inline app/views/components/presentable/details_component/_links.html.haml:8]: comment about refactor after release as in sidebar_component/_links?

File: app/views/backoffice/services/_form.html.haml [hunk 6]
```diff
@@ -74,21 +74,40 @@
                     %i.fas.fa-chevron-up
         .collapse{ id: "marketing", "aria-labelledby": "marketing-header",
             "data-parent": "#accordion-form",
-            class: ("show" unless collapsed?(service, [:description, :tagline, :logo, :multimedia, :use_cases_url])) }
+            class: ("show" unless collapsed?(service, [:description, :tagline, :logo, :"link_multimedia_urls.url",
```
Comment (kmarszalek) [inline app/views/backoffice/services/_form.html.haml:77]: both fields missing at line 63

File: app/views/backoffice/providers/form/_marketing.html.haml [hunk 7]
```diff
@@ -13,14 +13,25 @@
             %i.fas.fa-chevron-up
 .collapse{ id: "marketing", "aria-labelledby": "marketing-header",
           "data-parent": "#accordion-form",
-          class: ("show" unless collapsed?(provider, [:description, :logo, :multimedia])) }
+          class: ("show" unless collapsed?(provider, [:description, :logo,
+          :link_multimedia_urls, :"link_multimedia_urls.url"])) }
```
Comment (kmarszalek) [inline app/views/backoffice/providers/form/_marketing.html.haml:17]: both fields missing at line 3

File: app/helpers/url_helper.rb [hunk 8]
```diff
@@ -20,8 +20,12 @@ def self.url_valid?(url)
     return false unless uri.is_a?(URI::HTTP)
 
     response = Faraday.get(url)
-    response.status == 200
+    response.status == 200 || 301
```
Comment (kmarszalek) [inline app/helpers/url_helper.rb:23]: why 301?

**Review by kmarszalek — CHANGES_REQUESTED** [review]
looks fine, some minors, one question. + rebase :)

File: app/helpers/url_helper.rb (hunk 8 above)
Comment (goreck888) [reply app/helpers/url_helper.rb:23]: Removed.

**Pull request merged**